packages/experimental/src/components/APIKeyModal/APIKeyModal.stories.js

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright IBM Corp. 2020, 2020
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
+import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
+import { Button } from 'carbon-components-react';
+import { APIKeyModal } from '.';
+import styles from './_storybook-styles.scss'; // import index in case more files are added later.
+import mdx from './APIKeyModal.mdx';
+
+export default {
+  title: 'Experimental/APIKeyModal',
+  component: APIKeyModal,
+  parameters: {
+    styles,
+    docs: {
+      page: mdx,
+    },
+  },
+};
+
+const defaultProps = {
+  apiKey: '123-456-789',
+  apiKeyInputId: 'apiKeyInput',
+  apiKeyLabel: 'API key',
+  copyButtonText: 'Copy',
+  open: true,
+  secondaryButtonText: 'Close',
+  successBody: (
+    <p>
+      This is your unique API key and is non-recoverable. If you lose this API
+      key, you will have to reset it.
+    </p>
+  ),
+  successHeader: 'API key successully created',
+};
+
+const Template = (args) => {
+  return <APIKeyModal {...args} />;
+};
+
+/**
+ * Simple state manager for modals.
+ */
+/* eslint-disable react/prop-types */
+const ModalStateManager = ({
+  renderLauncher: LauncherContent,
+  children: ModalContent,
+}) => {
+  const [open, setOpen] = useState(false);
+  const [apiKey, setApiKey] = useState('');
+  const [loading, setLoading] = useState(false);
+  // eslint-disable-next-line
+  const setKeyHandler = async (name) => {
+    setLoading(true);
+    const timeout = () => new Promise((resolve) => setTimeout(resolve, 2000));
+    await timeout();
+    setApiKey('111-111-111-111');
+    setLoading(false);
+  };
+  return (
+    <>
+      {!ModalContent || typeof document === 'undefined'
+        ? null
+        : ReactDOM.createPortal(
```

Comment:
Using a portal should not be necessary to make the modal appear over other content.

Reply:
the portal is necessary for the modal component to show up properly in the docs section. if you don't it causes issues with the way the modal is rendered due to styling take a look here.

<img width="858" alt="Screen Shot 2020-11-17 at 2 25 08 PM" src="https://user-images.githubusercontent.com/6370760/99443682-c05c6c80-28e0-11eb-8c06-2a19b3430b15.png">


this is also how the modal should be rendered according to carbon and generally how react recommends rendering a modal component. this code should be more realistic in terms of how it's most likely to actually be consumed. i was able to rework the way the modal story code works though so now it should be able to use controls.

Reply:
@davidmenendez the narrow tearsheet, produced by @moores2, without the use of a portal. 

Can you swap notes and figure out the best route forward.